indexer/indexer.go

```diff
@@ -253,3 +281,37 @@ func (i *Indexer) collect() {
 		i.height++
 	}
 }
+
+// computeStartHeight decides the effective starting height given
+// - the next height after the last block in DB (dbNext)
+// - the current chain head (chainHead)
+// - whether a start height is configured (startSet)
+// - whether the configured value is 'latest' (latest)
+// - the configured numeric start value (startVal)
+// It enforces: effective >= dbNext and effective <= chainHead.
+func computeStartHeight(dbNext, chainHead int64, startSet bool, latest bool, startVal int64) int64 {
```

Comment:
We can make computeStartHeight clearer and safer by clamping in a single return using Go 1.23 built-in min/max.
```suggestion
func computeStartHeight(dbNext, chainHead int64, startSet bool, latest bool, startVal int64) int64 {
	if !startSet {
		return dbNext
	}

	var desired int64
	if latest {
		desired = chainHead
	} else {
		desired = max(startVal, 0)
	}

	return min(max(desired, dbNext), chainHead)
}
```